File: ipyvuedraggable/Draggable.py
from traitlets import Unicode, Dict
from ipyvue import VueWidget
from ._version import __version__

class Draggable(VueWidget):

    _model_name = Unicode('DraggableModel').tag(sync=True)

    _model_module = Unicode('jupyter-vuedraggable').tag(sync=True)

    _model_module_version = Unicode(__version__).tag(sync=True)

    group = Dict().tag(sync=True)

    tag = Unicode().tag(sync=True)

    component_data = Dict().tag(sync=True)


__all__ = ['Draggable']

File: js/src/Draggable.js
/* eslint camelcase: off */
import { VueModel } from 'jupyter-vue';
import { version } from "../package.json";

export class DraggableModel extends VueModel {
    defaults() {
        return {
            ...super.defaults(),
            ...{
                _model_name: 'DraggableModel',
                _model_module: 'jupyter-vuedraggable',
                _model_module_version: version,
                group: null,
                tag: null,
                component_data: null,
            },
        };
    }

    getVueTag() {
        return 'draggable';
    }
}

DraggableModel.serializers = {
    ...VueModel.serializers,
};
